cmd/bucket-notification-handlers.go

```diff
@@ -226,23 +226,14 @@ func writeNotification(w http.ResponseWriter, notification map[string][]Notifica
 // CRLF character used for chunked transfer in accordance with HTTP standards.
 var crlf = []byte("\r\n")
 
-// sendBucketNotification - writes notification back to client on the response writer
-// for each notification input, otherwise writes whitespace characters periodically
-// to keep the connection active. Each notification messages are terminated by CRLF
-// character. Upon any error received on response writer the for loop exits.
-func sendBucketNotification(w http.ResponseWriter, arnListenerCh <-chan []NotificationEvent) {
-	var dummyEvents = map[string][]NotificationEvent{"Records": nil}
-	// Continuously write to client either timely empty structures
-	// every 5 seconds, or return back the notifications.
+// keepListenerAlive writes empty records to ResponseWriter every
+// `globalSNSConnAlive` seconds to keep the connection alive.
+func keepListenerAlive(w http.ResponseWriter) {
+	var emptyEvent = map[string][]NotificationEvent{"Records": nil}
 	for {
 		select {
-		case events := <-arnListenerCh:
-			if err := writeNotification(w, map[string][]NotificationEvent{"Records": events}); err != nil {
-				errorIf(err, "Unable to write notification to client.")
-				return
-			}
-		case <-time.After(globalSNSConnAlive): // Wait for global conn active seconds.
-			if err := writeNotification(w, dummyEvents); err != nil {
+		case <-time.After(globalSNSConnAlive):
+			if err := writeNotification(w, emptyEvent); err != nil {
```

Comment:
It is not clear to me if `ResponseWriter` is concurrent-safe. We are writing in one thread here, and in another when events are sent.